nautobot_ssot_infoblox/utils/client.py

```diff
@@ -251,27 +279,83 @@ def get_all_ipv4address_networks(self, prefixes):
             }
         ]
         """
+
+        def get_ipaddrs(url_path: str, data: dict) -> list:
+            """Retrieve IP addresses specified in data payload.
+
+            Args:
+                url_path (str): The URL path to the API endpoint for requests.
+                data (dict): The data payload query of IP Addresses.
+
+            Returns:
+                list: List of dicts of IP Addresses for the specified prefixes or an empty list if no response.
+            """
+            response = None
+            try:
+                response = self._request(method="POST", path=url_path, data=data)
+            except HTTPError as err:
+                logger.info(err.response.text)
+            if response and len(response.json()) > 0:
+                logger.debug(response.json()[0])
+                return response.json()[0]
+            return []
+
+        def create_payload(prefix: str, view: str) -> dict:
+            """Create the payload structure for querying IP Addresses from subnets.
+
+            Args:
+                prefix (str): The prefix to get IP addresses for.
+                view (str): The Network View of the prefix being queried.
+
+            Returns:
+                dict: Dictionary containing query parameters for IP addresses from a prefix intended to be part of a list sent to request endpoint.
+            """
+            query = {
+                "method": "GET",
+                "object": "ipv4address",
+                "data": {"network_view": view, "network": prefix, "status": "USED"},
+                "args": {
+                    "_return_fields": "ip_address,mac_address,names,network,objects,status,types,usage,comment,extattrs"
+                },
+            }
+            return query
+
         url_path = "request"
-        payload = []
+        payload, ipaddrs = [], []
+        num_hosts = 0
         for prefix in prefixes:
-            payload.append(
-                {
-                    "method": "GET",
-                    "object": "ipv4address",
-                    "data": {"network_view": prefix[1], "network": prefix[0], "status": "USED"},
-                    "args": {
-                        "_return_fields": "ip_address,mac_address,names,network,objects,status,types,usage,comment,extattrs"
-                    },
-                }
-            )
-        data = json.dumps(payload)
-        try:
-            response = self._request(method="POST", path=url_path, data=data)
-        except HTTPError as err:
-            logger.info(err.response.text)
-            return []
-        logger.info(response.json()[0])
-        return response.json()[0]
+            view = prefix[1]
+            network = ipaddress.ip_network(prefix[0])
+            # make call with individual network if it's larger than 1000 hosts
```

Comment:
why? should this be a configurable option?

Reply:
Infoblox has a max_result of 1000 results unless you specify a different limit in the parameters. You can find details here: https://ipam.illinois.edu/wapidoc/index.html#get. "The number of objects returned is limited by the option _max_results or, if _max_results is not specified, 1000 objects. If _max_results is not specified, the appliance returns an error when the number of returned objects would exceed 1000. Similarly, if _max_results is set to -500 (maximum of 500 objecs) the appliance returns an error if the number of returned objects would exceed 500."

Reply:
maybe then add a note to avoid someone else asking the same in the future?

Reply:
Done.